sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -363,6 +370,15 @@ private[client] class Shim_v0_12 extends Shim with Logging {
     conf.getIntVar(HiveConf.ConfVars.METASTORE_CLIENT_CONNECT_RETRY_DELAY) * 1000L
   }
 
+  override def getTablesByType(
+      hive: Hive,
+      dbName: String,
+      pattern: String,
+      tableType: TableType): Seq[String] = {
+    throw new UnsupportedOperationException("Hive 2.2 and lower versions don't support " +
```

Comment:
Shall we fallback to `listTables` and then filter by table type? It's weird if SHOW VIEWS only works for hive 2.3+

Reply:
Yea, before I notice the API `getTablesByType`, I was planning to implement it by that way. But for `HiveExternalCatalog`, calling `HiveClient.getTableOption` for each table would be too heavy, right?  Maybe we can fallback to `listTables+filter` way if `getTablesByType` not supported. And for hive 2.3+, we use the efficient way?  Thanks!

Reply:
Yea, I'll implement that way instead of throwing error. Thanks!

Reply:
Can we implement without adding the method in `HiveShim` first, and improve with using `listTables`? Seems like it's not a must given my reading. Adding a method here requires to test in multiple profiles with multiple versions.